spec/features/registration_spec.rb

```diff
@@ -0,0 +1,57 @@
+feature 'registration' do
+  scenario 'a user can sign up' do
+    visit '/users/new'
+    fill_in('email', with: 'test@example.com')
+    fill_in('password', with: 'password123')
+    fill_in('name', with: 'Test Name')
+    fill_in('username', with: 'testuser1')
+    click_button('Sign Up')
+
+    expect(page).to have_content "Welcome, Test Name (@testuser1)"
+  end
+
+  scenario 'new user username must be unique' do
+    User.create(email: 'test@example.com', password: 'password123', name: 'Test Name', username: 'testuser1')
+
+    visit '/users/new'
+    fill_in('email', with: 'test3@example.com')
```

Comment:
Not extremely necessary however could refactor some of the re-used code into a helper method to clean-up tests. However not 100% needed as stated